src/main/java/innovimax/mixthem/operation/DefaultLineJoining.java

```diff
@@ -16,40 +17,70 @@
+  } 
 
-	@Override
-	public String process(String line1, String line2) throws MixException, ProcessException {
-		if (line1 == null || line2 == null) {
-			throw new ProcessException();
-		}
-		String join = null;
-		int col1 = 1;
-		int col2 = 1;
-		if (this.params.containsKey(RuleParam._JOIN_COL1)) {
-			col1 = this.params.get(RuleParam._JOIN_COL1).asInt();
-			col2 = col1;
-		}
-		if (this.params.containsKey(RuleParam._JOIN_COL2)) {
-			col2 = this.params.get(RuleParam._JOIN_COL2).asInt();			
-		}
-		List<String> list1 = Arrays.asList(line1.split("\\s"));
-		List<String> list2 = Arrays.asList(line2.split("\\s"));		
-		if (list1.size() >= col1 && list2.size() >= col2 && list1.get(col1 - 1).equals(list2.get(col2 - 1))) {
-			String part1 = list1.get(col1 - 1);
-			String part2 = list1.stream().filter(s -> !s.equals(part1)).collect(Collectors.joining(" "));
-			String part3 = list2.stream().filter(s -> !s.equals(part1)).collect(Collectors.joining(" "));
-			join = part1 + " " + part2 + " " + part3;				
-		}		
-		return join;
-	}
+  @Override
+  public void process(String line1, String line2, LineResult result) throws MixException {
+    result.reset();
+    if (line1 != null && line2 != null) {
+      List<String> list1 = Arrays.asList(line1.split(CellOperation.DEFAULT_SPLIT_CELL_REGEX.toString()));
+      List<String> list2 = Arrays.asList(line2.split(CellOperation.DEFAULT_SPLIT_CELL_REGEX.toString()));   
+      String cell1 = list1.size() >= this.col1 ? list1.get(this.col1 - 1) : null;
+      String cell2 = list2.size() >= this.col2 ? list2.get(this.col2 - 1) : null;
+      if (cell1 != null && cell2 != null) {
+        /*List<String> prevList1 = result.hasFirstLine() ?
+                        Arrays.asList(result.getFirstLine().split(CellOperation.DEFAULT_SPLIT_CELL_REGEX.toString())) :
+                        Collections.emptyList();
+        List<String> prevList2 = result.hasSecondLine() ?
+                        Arrays.asList(result.getSecondLine().split(CellOperation.DEFAULT_SPLIT_CELL_REGEX.toString())) :
+                        Collections.emptyList();    
+        String prevCell1 = prevList1.size() >= this.col1 ? prevList1.get(this.col1 - 1) : null;
+        String prevCell2 = prevList2.size() >= this.col2 ? prevList2.get(this.col2 - 1) : null;
+        if (cell1.equals(prevCell1)) {
+          joinLines(list1, prevList2, result);
+          result.preserveSecondLine();
+          result.setFirstLine(line1);
+        } else if (cell2.equals(prevCell2)) {
+          joinLines(prevList1, list2, result);
+          result.preserveFirstLine();
+          result.setSecondLine(line2);
+        } else {*/
+          switch (cell1.compareTo(cell2)) {
+            case 0:
+              joinLines(list1, list2, result);            
+              break;
+            case 1:           
+              result.preserveFirstLine();
+              break;
+            default:            
```

Comment:
compareTo can return any positive value (not only 1) : you must use https://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html#signum(int) to be sure it only has 3 values (-1, 0 and 1)

Reply:
done